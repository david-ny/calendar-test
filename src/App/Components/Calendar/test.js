import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Calendar from './index';
import { configTestData } from '../../test/testData';

const props = configTestData;

describe('Calendar', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Calendar {...props} />, div);
  });
  test('has a valid snapshot', () => {
    const component = renderer.create(
      <Calendar {...props} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
