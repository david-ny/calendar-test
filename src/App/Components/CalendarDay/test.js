import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import CalendarDay from './index';
import { CalendarDayTestData } from '../../test/testData';

const props = CalendarDayTestData;

describe('CalendarDay', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<CalendarDay {...props} />, div);
  });
  test('has a valid snapshot', () => {
    const component = renderer.create(
      <CalendarDay {...props} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
