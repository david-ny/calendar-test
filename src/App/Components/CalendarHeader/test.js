import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import CalendarHeader from './index';
import { configTestData } from '../../test/testData';

const props = configTestData;

describe('CalendarHeader', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<CalendarHeader {...props} />, div);
  });
  test('has a valid snapshot', () => {
    const component = renderer.create(
      <CalendarHeader {...props} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
