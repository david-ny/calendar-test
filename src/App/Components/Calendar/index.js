import React, { Component } from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import CalendarHeader from '../CalendarHeader';
import CalendarDay from '../CalendarDay';


class Calendar extends Component {
  constructor(props) {
    super(props);
    const { locale } = this.props;
    moment.locale(locale);
  }

  render() {
    const { dateString, locale } = this.props;
    // const { dateString } = this.props;
    const validatedDateString = !moment(dateString).isValid()
      ? moment().format('YYYY-MM-DD')
      : dateString;

    moment.locale(locale);
    const numberOfDaysInPrevMonth = moment(validatedDateString).subtract(1, 'months').daysInMonth();
    const numberOfDaysInMonth = moment(validatedDateString).daysInMonth();
    const numberOfDaysInNextMonth = moment(validatedDateString).add(1, 'months').daysInMonth();
    const firstDayInMonth = moment(`${validatedDateString.slice(0, 7)}-01`).weekday();
    const prevMonthArr = Array.from(Array(numberOfDaysInPrevMonth + 1).keys())
      .slice(1, numberOfDaysInPrevMonth + 1)
      .slice(
        (numberOfDaysInPrevMonth + 1) - (firstDayInMonth + 1),
        numberOfDaysInPrevMonth,
      );
    const currMonthArr = Array.from(Array(numberOfDaysInMonth + 1).keys())
      .slice(1, numberOfDaysInMonth + 1);
    const nextMonthArr = Array.from(Array(numberOfDaysInNextMonth + 1).keys())
      .slice(1, numberOfDaysInNextMonth + 1)
      .slice(
        0,
        (42 - (prevMonthArr.length + numberOfDaysInMonth)),
      );

    const prevMonthDaysArr = prevMonthArr.map(item =>
      (<CalendarDay
        key={`0${item}-prev`}
        dayNum={item}
        dateString={validatedDateString}
        locale={locale}
        prev
      />));
    const currMonthDaysArr = currMonthArr.map(item =>
      (<CalendarDay
        key={`0${item}`}
        dayNum={item}
        dateString={validatedDateString}
        locale={locale}
      />));
    const nextMonthDaysArr = nextMonthArr.map(item =>
      (<CalendarDay
        key={`0${item}-next`}
        dayNum={item}
        dateString={validatedDateString}
        locale={locale}
        next
      />));

    const calendarDays = [
      ...prevMonthDaysArr,
      ...currMonthDaysArr,
      ...nextMonthDaysArr,
    ];

    return (
      <div className="calendar">
        <div style={{
          display: 'flex',
          alignItems: 'flex-end',
         }}
        >
          <div style={{ flexGgrow: '1', width: '100%' }}>
            <CalendarHeader locale={locale} />
            <div style={{ display: 'flex', flexWrap: 'wrap' }}>
              {calendarDays.map(item => item)}
            </div>
          </div>
        </div>
      </div>
    );
  }
}

Calendar.propTypes = {
  dateString: PropTypes.string,
  locale: PropTypes.string,
};
Calendar.defaultProps = {
  dateString: moment().format('YYYY-MM-DD'),
  locale: 'en',
};

export default Calendar;
