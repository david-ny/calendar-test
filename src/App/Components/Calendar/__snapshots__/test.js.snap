// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar has a valid snapshot 1`] = `
<div
  className="calendar"
>
  <div
    style={
      Object {
        "alignItems": "flex-end",
        "display": "flex",
      }
    }
  >
    <div
      style={
        Object {
          "flexGgrow": "1",
          "width": "100%",
        }
      }
    >
      <div
        className="calendar-header"
        style={
          Object {
            "display": "flex",
          }
        }
      >
        <div
          style={
            Object {
              "flexBasis": "14.2857%",
              "maxWidth": "14.2857%",
              "overflow": "hidden",
            }
          }
        >
          Su
        </div>
        <div
          style={
            Object {
              "flexBasis": "14.2857%",
              "maxWidth": "14.2857%",
              "overflow": "hidden",
            }
          }
        >
          Mo
        </div>
        <div
          style={
            Object {
              "flexBasis": "14.2857%",
              "maxWidth": "14.2857%",
              "overflow": "hidden",
            }
          }
        >
          Tu
        </div>
        <div
          style={
            Object {
              "flexBasis": "14.2857%",
              "maxWidth": "14.2857%",
              "overflow": "hidden",
            }
          }
        >
          We
        </div>
        <div
          style={
            Object {
              "flexBasis": "14.2857%",
              "maxWidth": "14.2857%",
              "overflow": "hidden",
            }
          }
        >
          Th
        </div>
        <div
          style={
            Object {
              "flexBasis": "14.2857%",
              "maxWidth": "14.2857%",
              "overflow": "hidden",
            }
          }
        >
          Fr
        </div>
        <div
          style={
            Object {
              "flexBasis": "14.2857%",
              "maxWidth": "14.2857%",
              "overflow": "hidden",
            }
          }
        >
          Sa
        </div>
      </div>
      <div
        style={
          Object {
            "display": "flex",
            "flexWrap": "wrap",
          }
        }
      >
        <button
          className=""
          style={
            Object {
              "color": "red",
              "flexBasis": "14.2857%",
              "maxWidth": "14.2857%",
              "overflow": "hidden",
            }
          }
          type="button"
        >
          <time
            dateTime="2018-04-29 00:04"
            style={
              Object {
                "opacity": 0.3,
              }
            }
          >
            29
          </time>
        </button>
        <button
          className=""
          style={
            Object {
              "color": "gray",
              "flexBasis": "14.2857%",
              "maxWidth": "14.2857%",
              "overflow": "hidden",
            }
          }
          type="button"
        >
          <time
            dateTime="2018-04-30 00:04"
            style={
              Object {
                "opacity": 0.3,
              }
            }
          >
            30
          </time>
        </button>
        <button
          className=""
          style={
            Object {
              "color": "gray",
              "flexBasis": "14.2857%",
              "maxWidth": "14.2857%",
              "overflow": "hidden",
            }
          }
          type="button"
        >
          <time
            dateTime="2018-05-01 00:05"
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            1
          </time>
        </button>
        <button
          className=""
          style={
            Object {
              "color": "gray",
              "flexBasis": "14.2857%",
              "maxWidth": "14.2857%",
              "overflow": "hidden",
            }
          }
          type="button"
        >
          <time
            dateTime="2018-05-02 00:05"
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            2
          </time>
        </button>
        <button
          className=""
          style={
            Object {
              "color": "gray",
              "flexBasis": "14.2857%",
              "maxWidth": "14.2857%",
              "overflow": "hidden",
            }
          }
          type="button"
        >
          <time
            dateTime="2018-05-03 00:05"
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            3
          </time>
        </button>
        <button
          className=""
          style={
            Object {
              "color": "gray",
              "flexBasis": "14.2857%",
              "maxWidth": "14.2857%",
              "overflow": "hidden",
            }
          }
          type="button"
        >
          <time
            dateTime="2018-05-04 00:05"
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            4
          </time>
        </button>
        <button
          className=""
          style={
            Object {
              "color": "red",
              "flexBasis": "14.2857%",
              "maxWidth": "14.2857%",
              "overflow": "hidden",
            }
          }
          type="button"
        >
          <time
            dateTime="2018-05-05 00:05"
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            5
          </time>
        </button>
        <button
          className=""
          style={
            Object {
              "color": "red",
              "flexBasis": "14.2857%",
              "maxWidth": "14.2857%",
              "overflow": "hidden",
            }
          }
          type="button"
        >
          <time
            dateTime="2018-05-06 00:05"
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            6
          </time>
        </button>
        <button
          className=""
          style={
            Object {
              "color": "gray",
              "flexBasis": "14.2857%",
              "maxWidth": "14.2857%",
              "overflow": "hidden",
            }
          }
          type="button"
        >
          <time
            dateTime="2018-05-07 00:05"
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            7
          </time>
        </button>
        <button
          className=""
          style={
            Object {
              "color": "gray",
              "flexBasis": "14.2857%",
              "maxWidth": "14.2857%",
              "overflow": "hidden",
            }
          }
          type="button"
        >
          <time
            dateTime="2018-05-08 00:05"
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            8
          </time>
        </button>
        <button
          className=""
          style={
            Object {
              "color": "gray",
              "flexBasis": "14.2857%",
              "maxWidth": "14.2857%",
              "overflow": "hidden",
            }
          }
          type="button"
        >
          <time
            dateTime="2018-05-09 00:05"
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            9
          </time>
        </button>
        <button
          className=""
          style={
            Object {
              "color": "gray",
              "flexBasis": "14.2857%",
              "maxWidth": "14.2857%",
              "overflow": "hidden",
            }
          }
          type="button"
        >
          <time
            dateTime="2018-05-10 00:05"
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            10
          </time>
        </button>
        <button
          className="today"
          style={
            Object {
              "color": "gray",
              "flexBasis": "14.2857%",
              "maxWidth": "14.2857%",
              "overflow": "hidden",
            }
          }
          type="button"
        >
          <time
            dateTime="2018-05-11 00:05"
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            11
          </time>
        </button>
        <button
          className=""
          style={
            Object {
              "color": "red",
              "flexBasis": "14.2857%",
              "maxWidth": "14.2857%",
              "overflow": "hidden",
            }
          }
          type="button"
        >
          <time
            dateTime="2018-05-12 00:05"
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            12
          </time>
        </button>
        <button
          className=""
          style={
            Object {
              "color": "red",
              "flexBasis": "14.2857%",
              "maxWidth": "14.2857%",
              "overflow": "hidden",
            }
          }
          type="button"
        >
          <time
            dateTime="2018-05-13 00:05"
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            13
          </time>
        </button>
        <button
          className=""
          style={
            Object {
              "color": "gray",
              "flexBasis": "14.2857%",
              "maxWidth": "14.2857%",
              "overflow": "hidden",
            }
          }
          type="button"
        >
          <time
            dateTime="2018-05-14 00:05"
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            14
          </time>
        </button>
        <button
          className=""
          style={
            Object {
              "color": "gray",
              "flexBasis": "14.2857%",
              "maxWidth": "14.2857%",
              "overflow": "hidden",
            }
          }
          type="button"
        >
          <time
            dateTime="2018-05-15 00:05"
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            15
          </time>
        </button>
        <button
          className=""
          style={
            Object {
              "color": "gray",
              "flexBasis": "14.2857%",
              "maxWidth": "14.2857%",
              "overflow": "hidden",
            }
          }
          type="button"
        >
          <time
            dateTime="2018-05-16 00:05"
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            16
          </time>
        </button>
        <button
          className=""
          style={
            Object {
              "color": "gray",
              "flexBasis": "14.2857%",
              "maxWidth": "14.2857%",
              "overflow": "hidden",
            }
          }
          type="button"
        >
          <time
            dateTime="2018-05-17 00:05"
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            17
          </time>
        </button>
        <button
          className=""
          style={
            Object {
              "color": "gray",
              "flexBasis": "14.2857%",
              "maxWidth": "14.2857%",
              "overflow": "hidden",
            }
          }
          type="button"
        >
          <time
            dateTime="2018-05-18 00:05"
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            18
          </time>
        </button>
        <button
          className=""
          style={
            Object {
              "color": "red",
              "flexBasis": "14.2857%",
              "maxWidth": "14.2857%",
              "overflow": "hidden",
            }
          }
          type="button"
        >
          <time
            dateTime="2018-05-19 00:05"
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            19
          </time>
        </button>
        <button
          className=""
          style={
            Object {
              "color": "red",
              "flexBasis": "14.2857%",
              "maxWidth": "14.2857%",
              "overflow": "hidden",
            }
          }
          type="button"
        >
          <time
            dateTime="2018-05-20 00:05"
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            20
          </time>
        </button>
        <button
          className=""
          style={
            Object {
              "color": "gray",
              "flexBasis": "14.2857%",
              "maxWidth": "14.2857%",
              "overflow": "hidden",
            }
          }
          type="button"
        >
          <time
            dateTime="2018-05-21 00:05"
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            21
          </time>
        </button>
        <button
          className=""
          style={
            Object {
              "color": "gray",
              "flexBasis": "14.2857%",
              "maxWidth": "14.2857%",
              "overflow": "hidden",
            }
          }
          type="button"
        >
          <time
            dateTime="2018-05-22 00:05"
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            22
          </time>
        </button>
        <button
          className=""
          style={
            Object {
              "color": "gray",
              "flexBasis": "14.2857%",
              "maxWidth": "14.2857%",
              "overflow": "hidden",
            }
          }
          type="button"
        >
          <time
            dateTime="2018-05-23 00:05"
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            23
          </time>
        </button>
        <button
          className=""
          style={
            Object {
              "color": "gray",
              "flexBasis": "14.2857%",
              "maxWidth": "14.2857%",
              "overflow": "hidden",
            }
          }
          type="button"
        >
          <time
            dateTime="2018-05-24 00:05"
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            24
          </time>
        </button>
        <button
          className=""
          style={
            Object {
              "color": "gray",
              "flexBasis": "14.2857%",
              "maxWidth": "14.2857%",
              "overflow": "hidden",
            }
          }
          type="button"
        >
          <time
            dateTime="2018-05-25 00:05"
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            25
          </time>
        </button>
        <button
          className=""
          style={
            Object {
              "color": "red",
              "flexBasis": "14.2857%",
              "maxWidth": "14.2857%",
              "overflow": "hidden",
            }
          }
          type="button"
        >
          <time
            dateTime="2018-05-26 00:05"
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            26
          </time>
        </button>
        <button
          className=""
          style={
            Object {
              "color": "red",
              "flexBasis": "14.2857%",
              "maxWidth": "14.2857%",
              "overflow": "hidden",
            }
          }
          type="button"
        >
          <time
            dateTime="2018-05-27 00:05"
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            27
          </time>
        </button>
        <button
          className=""
          style={
            Object {
              "color": "gray",
              "flexBasis": "14.2857%",
              "maxWidth": "14.2857%",
              "overflow": "hidden",
            }
          }
          type="button"
        >
          <time
            dateTime="2018-05-28 00:05"
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            28
          </time>
        </button>
        <button
          className=""
          style={
            Object {
              "color": "gray",
              "flexBasis": "14.2857%",
              "maxWidth": "14.2857%",
              "overflow": "hidden",
            }
          }
          type="button"
        >
          <time
            dateTime="2018-05-29 00:05"
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            29
          </time>
        </button>
        <button
          className=""
          style={
            Object {
              "color": "gray",
              "flexBasis": "14.2857%",
              "maxWidth": "14.2857%",
              "overflow": "hidden",
            }
          }
          type="button"
        >
          <time
            dateTime="2018-05-30 00:05"
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            30
          </time>
        </button>
        <button
          className=""
          style={
            Object {
              "color": "gray",
              "flexBasis": "14.2857%",
              "maxWidth": "14.2857%",
              "overflow": "hidden",
            }
          }
          type="button"
        >
          <time
            dateTime="2018-05-31 00:05"
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            31
          </time>
        </button>
        <button
          className=""
          style={
            Object {
              "color": "gray",
              "flexBasis": "14.2857%",
              "maxWidth": "14.2857%",
              "overflow": "hidden",
            }
          }
          type="button"
        >
          <time
            dateTime="2018-06-01 00:06"
            style={
              Object {
                "opacity": 0.3,
              }
            }
          >
            1
          </time>
        </button>
        <button
          className=""
          style={
            Object {
              "color": "red",
              "flexBasis": "14.2857%",
              "maxWidth": "14.2857%",
              "overflow": "hidden",
            }
          }
          type="button"
        >
          <time
            dateTime="2018-06-02 00:06"
            style={
              Object {
                "opacity": 0.3,
              }
            }
          >
            2
          </time>
        </button>
        <button
          className=""
          style={
            Object {
              "color": "red",
              "flexBasis": "14.2857%",
              "maxWidth": "14.2857%",
              "overflow": "hidden",
            }
          }
          type="button"
        >
          <time
            dateTime="2018-06-03 00:06"
            style={
              Object {
                "opacity": 0.3,
              }
            }
          >
            3
          </time>
        </button>
        <button
          className=""
          style={
            Object {
              "color": "gray",
              "flexBasis": "14.2857%",
              "maxWidth": "14.2857%",
              "overflow": "hidden",
            }
          }
          type="button"
        >
          <time
            dateTime="2018-06-04 00:06"
            style={
              Object {
                "opacity": 0.3,
              }
            }
          >
            4
          </time>
        </button>
        <button
          className=""
          style={
            Object {
              "color": "gray",
              "flexBasis": "14.2857%",
              "maxWidth": "14.2857%",
              "overflow": "hidden",
            }
          }
          type="button"
        >
          <time
            dateTime="2018-06-05 00:06"
            style={
              Object {
                "opacity": 0.3,
              }
            }
          >
            5
          </time>
        </button>
        <button
          className=""
          style={
            Object {
              "color": "gray",
              "flexBasis": "14.2857%",
              "maxWidth": "14.2857%",
              "overflow": "hidden",
            }
          }
          type="button"
        >
          <time
            dateTime="2018-06-06 00:06"
            style={
              Object {
                "opacity": 0.3,
              }
            }
          >
            6
          </time>
        </button>
        <button
          className=""
          style={
            Object {
              "color": "gray",
              "flexBasis": "14.2857%",
              "maxWidth": "14.2857%",
              "overflow": "hidden",
            }
          }
          type="button"
        >
          <time
            dateTime="2018-06-07 00:06"
            style={
              Object {
                "opacity": 0.3,
              }
            }
          >
            7
          </time>
        </button>
        <button
          className=""
          style={
            Object {
              "color": "gray",
              "flexBasis": "14.2857%",
              "maxWidth": "14.2857%",
              "overflow": "hidden",
            }
          }
          type="button"
        >
          <time
            dateTime="2018-06-08 00:06"
            style={
              Object {
                "opacity": 0.3,
              }
            }
          >
            8
          </time>
        </button>
        <button
          className=""
          style={
            Object {
              "color": "red",
              "flexBasis": "14.2857%",
              "maxWidth": "14.2857%",
              "overflow": "hidden",
            }
          }
          type="button"
        >
          <time
            dateTime="2018-06-09 00:06"
            style={
              Object {
                "opacity": 0.3,
              }
            }
          >
            9
          </time>
        </button>
      </div>
    </div>
  </div>
</div>
`;
